cmd/cloud-controller-manager/gkeservicecontroller.go

```diff
@@ -0,0 +1,50 @@
+package main
+
+import (
+	"context"
+
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	cloudprovider "k8s.io/cloud-provider"
+	"k8s.io/cloud-provider-gcp/cmd/cloud-controller-manager/options"
+	gkeservicecontroller "k8s.io/cloud-provider-gcp/pkg/controller/service"
+	"k8s.io/cloud-provider/app"
+	cloudcontrollerconfig "k8s.io/cloud-provider/app/config"
+	controllermanagerapp "k8s.io/controller-manager/app"
+	"k8s.io/controller-manager/controller"
+	"k8s.io/klog/v2"
+)
+
+type gkeServiceController struct {
+	gkeServiceControllerOptions options.GkeServiceControllerOptions
+}
+
+// startGkeServiceControllerWrapper is used to take cloud config as input and start the GKE service controller
+func (gkeServiceController *gkeServiceController) startGkeServiceControllerWrapper(initContext app.ControllerInitContext, completedConfig *cloudcontrollerconfig.CompletedConfig, cloud cloudprovider.Interface) app.InitFunc {
+	if gkeServiceController.gkeServiceControllerOptions.EnableGkeServiceController {
+		return func(ctx context.Context, controllerContext controllermanagerapp.ControllerContext) (controller.Interface, bool, error) {
+			return startGkeServiceController(ctx, initContext, controllerContext, completedConfig, cloud)
+		}
+	}
+	return app.StartServiceControllerWrapper(initContext, completedConfig, cloud)
```

Comment:
users do (example) `--controllers=-services,nodeipam` so this has to be controlled by the same mechanism, the controllers flag selects which controllers are enabled or disabled.

If we keep the existing controller and we just add this new controller to handle the new classes, this will look like
`--controllers=services,nodeipam,gkeservices`

This guarantees the compatibility with the existing manifest and limits the blast radious of new functionality added on the new classes breaking existing kubernetes/default service controller